velox/functions/sparksql/Register.cpp

```diff
@@ -255,6 +255,9 @@ void registerFunctions(const std::string& prefix) {
   registerFunction<DayOfWeekFunction, int32_t, Date>(
       {prefix + "dow", prefix + "dayofweek"});
 
+  registerFunction<MonthFunction, int32_t, Timestamp>({prefix + "month"});
```

Comment:
Hi @JkSelf, It looks spark only accepts Date type for this function. For other types like Timestamp, I think it will do an implicit cast. So, we may just need to consider Date type here.

https://github.com/apache/spark/blob/b9ac7d38ea7f4f7e57c4bbb7b5e5bc836b2b1c58/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala#L479

Reply:
@PHILO-HE Thanks for your review. Updated.